web/web/auth/auth.py

```diff
@@ -9,5 +16,48 @@
 
 @auth_bp.route('/')
 @auth_bp.route('/auth', strict_slashes=False)
+@jwt_optional
 def index():
-    return render_template('auth/login.html')
+    access_token = request.cookies.get('access_token')
+    if access_token:
+        return redirect('power/capacity')
+    else:
+        return render_template('auth/login.html')
+
+
+# see this repo for JWT redis blacklist setup:
+# https://github.com/vimalloc/flask-jwt-extended/blob/master/examples/redis_blacklist.py
+
+@auth_bp.route('/auth/revoke', methods=['DELETE'])
+@jwt_required
```

Comment:
Can we move this to under the API folder?
Also, if we include the `@jwt_refresh_token_required` decorator wouldn't we bypass the need to get the refresh token explicitly from the cookie?

Also, may be worth explicitly setting some additional JWT config variables like `JWT_TOKEN_LOCATION`

Reply:
the jwt_refresh_token_required decorator looks for the token in the header, not in the cookies. I can only pass in one of the tokens in the authorization header, leaving the other one to be retrieved via cookies in the endpoint. Do you know of an alternate way to pass in an additional authorization header? Will look into the other configs

Reply:
No need to do this now since you're off. But how are you passing the token in the header?

Reply:
I am passing it in the headers with the "Authorization" key and the value is "Bearer {token}" 

`try {
            const accessTokenJson = {
                headers: {
                    Authorization: "Bearer " + Cookies.get("access_token"),
                },
            };
            api.delete(
                "logout",
                accessTokenJson,
                () => {
                    Cookies.remove("access_token");
                    Cookies.remove("refresh_token");
                    window.location.href = "/";
                    dispatch(completeLogout);
                },
                (error) => {
                    createErrorMessage("Error", "Unable to log out");
                }
            );
        } catch (error) {
            createErrorMessage("Server error", "Something went wrong");
        }
    };`

Reply:
ah got it - I think I missed that. I guess my thought was that we'd be able to deduce the user from the access token and then use the user information to get its cached refresh token and then expire it as well. That way only the access_token would need to be passed in the header.
Anyways, I think this piece looks good. Thanks for clarifying.